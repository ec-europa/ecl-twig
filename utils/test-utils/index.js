const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
