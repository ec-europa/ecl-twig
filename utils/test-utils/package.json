{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
