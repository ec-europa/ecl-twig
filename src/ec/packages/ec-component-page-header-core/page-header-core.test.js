import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

import {
  demoTitleContent,
  demoMetaTitleContent,
  demoMetaTitleDescriptionContent,
} from './demo/data';

const defaultSprite = '/static/icons.svg';

demoTitleContent.breadcrumb.icon_file_path = defaultSprite;
demoMetaTitleContent.breadcrumb.icon_file_path = defaultSprite;
demoMetaTitleDescriptionContent.breadcrumb.icon_file_path = defaultSprite;

describe('EC - Page Header Core', () => {
  const template =
    '@ecl-twig/ec-component-page-header-core/ecl-page-header-core.html.twig';
  const render = params => renderTwigFileAsNode(template, params);

  describe('title', () => {
    test(`- renders correctly`, () => {
      expect.assertions(1);
      return expect(render(demoTitleContent)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(demoTitleContent, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(demoTitleContent, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('meta-title', () => {
    test(`- renders correctly`, () => {
      expect.assertions(1);
      return expect(render(demoMetaTitleContent)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(demoMetaTitleContent, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(demoMetaTitleContent, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('meta-title-description', () => {
    test(`- renders correctly`, () => {
      expect.assertions(1);
      return expect(
        render(demoMetaTitleDescriptionContent)
      ).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(demoMetaTitleDescriptionContent, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(demoMetaTitleDescriptionContent, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
});
