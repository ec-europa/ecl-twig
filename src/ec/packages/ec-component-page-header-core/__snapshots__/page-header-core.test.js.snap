// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Page Header meta-title - renders correctly 1`] = `
<div
  class="ecl-page-header-core"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb-core ecl-page-header-core__breadcrumb"
      data-ecl-breadcrumb-core="true"
    >
      <ol
        class="ecl-breadcrumb-core__container"
      >
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          class="ecl-breadcrumb-core__segment ecl-breadcrumb-core__current-page"
          data-ecl-breadcrumb-core-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-core__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-core__title"
    >
      Page title
    </h1>
  </div>
</div>
`;

exports[`EC - Page Header meta-title renders correctly with extra attributes 1`] = `
<div
  class="ecl-page-header-core"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb-core ecl-page-header-core__breadcrumb"
      data-ecl-breadcrumb-core="true"
    >
      <ol
        class="ecl-breadcrumb-core__container"
      >
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          class="ecl-breadcrumb-core__segment ecl-breadcrumb-core__current-page"
          data-ecl-breadcrumb-core-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-core__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-core__title"
    >
      Page title
    </h1>
  </div>
</div>
`;

exports[`EC - Page Header meta-title renders correctly with extra class names 1`] = `
<div
  class="ecl-page-header-core custom-class custom-class--test"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb-core ecl-page-header-core__breadcrumb"
      data-ecl-breadcrumb-core="true"
    >
      <ol
        class="ecl-breadcrumb-core__container"
      >
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          class="ecl-breadcrumb-core__segment ecl-breadcrumb-core__current-page"
          data-ecl-breadcrumb-core-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-core__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-core__title"
    >
      Page title
    </h1>
  </div>
</div>
`;

exports[`EC - Page Header meta-title-description - renders correctly 1`] = `
<div
  class="ecl-page-header-core"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb-core ecl-page-header-core__breadcrumb"
      data-ecl-breadcrumb-core="true"
    >
      <ol
        class="ecl-breadcrumb-core__container"
      >
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          class="ecl-breadcrumb-core__segment ecl-breadcrumb-core__current-page"
          data-ecl-breadcrumb-core-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-core__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-core__title"
    >
      Page title
    </h1>
    <p
      class="ecl-page-header-core__description"
    >
      Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium
    </p>
  </div>
</div>
`;

exports[`EC - Page Header meta-title-description renders correctly with extra attributes 1`] = `
<div
  class="ecl-page-header-core"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb-core ecl-page-header-core__breadcrumb"
      data-ecl-breadcrumb-core="true"
    >
      <ol
        class="ecl-breadcrumb-core__container"
      >
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          class="ecl-breadcrumb-core__segment ecl-breadcrumb-core__current-page"
          data-ecl-breadcrumb-core-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-core__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-core__title"
    >
      Page title
    </h1>
    <p
      class="ecl-page-header-core__description"
    >
      Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium
    </p>
  </div>
</div>
`;

exports[`EC - Page Header meta-title-description renders correctly with extra class names 1`] = `
<div
  class="ecl-page-header-core custom-class custom-class--test"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb-core ecl-page-header-core__breadcrumb"
      data-ecl-breadcrumb-core="true"
    >
      <ol
        class="ecl-breadcrumb-core__container"
      >
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          class="ecl-breadcrumb-core__segment ecl-breadcrumb-core__current-page"
          data-ecl-breadcrumb-core-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-core__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-core__title"
    >
      Page title
    </h1>
    <p
      class="ecl-page-header-core__description"
    >
      Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium
    </p>
  </div>
</div>
`;

exports[`EC - Page Header title - renders correctly 1`] = `
<div
  class="ecl-page-header-core"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb-core ecl-page-header-core__breadcrumb"
      data-ecl-breadcrumb-core="true"
    >
      <ol
        class="ecl-breadcrumb-core__container"
      >
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          class="ecl-breadcrumb-core__segment ecl-breadcrumb-core__current-page"
          data-ecl-breadcrumb-core-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <h1
      class="ecl-page-header-core__title"
    >
      Page title
    </h1>
  </div>
</div>
`;

exports[`EC - Page Header title renders correctly with extra attributes 1`] = `
<div
  class="ecl-page-header-core"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb-core ecl-page-header-core__breadcrumb"
      data-ecl-breadcrumb-core="true"
    >
      <ol
        class="ecl-breadcrumb-core__container"
      >
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          class="ecl-breadcrumb-core__segment ecl-breadcrumb-core__current-page"
          data-ecl-breadcrumb-core-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <h1
      class="ecl-page-header-core__title"
    >
      Page title
    </h1>
  </div>
</div>
`;

exports[`EC - Page Header title renders correctly with extra class names 1`] = `
<div
  class="ecl-page-header-core custom-class custom-class--test"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb-core ecl-page-header-core__breadcrumb"
      data-ecl-breadcrumb-core="true"
    >
      <ol
        class="ecl-breadcrumb-core__container"
      >
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          class="ecl-breadcrumb-core__segment"
          data-ecl-breadcrumb-core-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb-core__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--2xs ecl-icon--rotate-90 ecl-breadcrumb-core__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          class="ecl-breadcrumb-core__segment ecl-breadcrumb-core__current-page"
          data-ecl-breadcrumb-core-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <h1
      class="ecl-page-header-core__title"
    >
      Page title
    </h1>
  </div>
</div>
`;
